### DISCUSSION TITLE
Einbindung von Powerfox

### DISCUSSION THREAD

**Discussion by norbertle** [solectrus/solectrus#3355] [Ideas]
Es wäre toll wenn man die Werte für 1.8.0 und 2.8.0, also Bezug und Einspeisung, am Stromzähler in Solectrus mit einlesen könnte über vermutlich einen neuen Connector / Container. Den Vorteil sehe ich darin das man Einspeisung und Bezug damit mit den Werten am Stromzähler vergleichen könnte. Vorteil ist wenn z. B. der Internetzugang weg ist und man deshalb über ein VPN nicht auf Solectros zugreifen kann man die Delta Werte über den Stromzähler erhält. 
Es stellt sich natürlich die Frage ob das Lizenztechnisch über die öffentliche API auch möglich wäre.

**Comment by ledermann** [comment]
So ganz habe ich nicht verstanden, was dir konkret vorschwebt.

Netzbezug (1.8.0) und Einspeisung (2.8.0) lassen sich, je nach Stromzähler, sicherlich separat auslesen. Für PowerFox gibt es eine [API](https://www.powerfox.energy/daten-von-powerfox-per-api-nutzen-2/), bei klassischen Zählern ließe sich mit IR-Lesegeräten und MQTT etwas basteln. Sicher gibt es es weitere Möglichkeiten.

Beide Messwerte erhält man üblicherweise auch über den Stromspeicher, wobei die Werte dann idealerweise mit dem Stromzähler übereinstimmen, aber theoretisch auch davon abweichen könnten.

Es stellt sich mir die Frage, was zu tun wäre, wenn es eine signifikante Abweichung gibt. SOLECTRUS benötigt eine eindeutige Angabe von Netzbezug und Einspeisung.

Dein Argument mit dem Internetausfall leuchtet mir auch nicht. Wenn ich z.B. von meinem SENEC Messwerte abrufe, dann funktioniert das auch bei Internetausfall - weil sowohl der Stromspeicher als auch mein SENEC-Collector lokal laufen. Würde ich hingegen eine öffentliche API abfragen (z.B. PowerFox) würde das bei Internetausfall nicht mehr möglich sein.

Auch ist mir nicht klar, an welcher Stelle ein VPN ins Spiel kommt. Du verwendest vielleicht eines, um von außerhalb auf dein lokal laufenden SOLECTRUS zugreifen zu können. Bei einem Internetausfall kommst du da natürlich nicht dran, aber lokal (im Haus) läuft ja alles weiter.
